src/internal-links/handler.js

```diff
@@ -106,7 +108,67 @@ export async function internalLinksAuditRunner(auditUrl, context, site) {
   };
 }
 
+async function convertToOpportunity(auditUrl, auditData, context) {
+  const {
+    dataAccess,
+    log,
+  } = context;
+
+  const opportunities = await dataAccess.Opportunity.allBySiteIdAndStatus(auditData.siteId, 'NEW');
+  let opportunity = opportunities.find((oppty) => oppty.getType() === AUDIT_TYPE);
+
+  if (!opportunity) {
+    const opportunityData = {
+      siteId: auditData.siteId,
+      auditId: auditData.id,
+      runbook: 'https://adobe.sharepoint.com/sites/aemsites-engineering/Shared%20Documents/3%20-%20Experience%20Success/SpaceCat/Runbooks/Experience_Success_Studio_Broken_Internal_Links_Runbook.docx?web=1',
+      type: AUDIT_TYPE,
+      origin: 'AUTOMATION',
+      title: 'Broken Internal Links',
+      description: 'Broken internal links can significantly harm website\'s SEO by preventing search engines from properly indexing site, potentially reducing search rankings',
+      guidance: {
+        steps: [
+          'Update each broken internal link to valid URLs.',
+          'Test the implemented changes manually to ensure they are working as expected.',
+          'Monitor internal links for 404 errors in RUM tool over time to ensure they are functioning correctly.',
+        ],
+      },
+      tags: [
+        'SEO',
+        'Web indexing',
+        'Web crawling',
+        'User Experience',
+        'Organic Traffic',
+      ],
+    };
+    opportunity = await dataAccess.Opportunity.create(opportunityData);
+  } else {
+    opportunity.setAuditId(auditData.id);
+    await opportunity.save();
+  }
+
+  const buildKey = (data) => `${data.siteId}|${data.auditResult.brokenInternalLinks.map((item) => item.url_to)};}`;
```

Comment:
I think, we should also use `url_from` in the key.
It is possible that 2 different pages are using the same broken link. These 2 are considered as separate entries, fwiu.

Reply:
This is good point, the combination of url_from and url_to should be better in this case, I think. PR updated.